cloud/deployment/deployment.go

```diff
@@ -719,28 +772,328 @@ func Update(deploymentID, label, ws, description, deploymentName, dagDeploy, exe
+			}
+		}
+		if defaultTaskPodMemory == "" {
+			if currentDeployment.DefaultTaskPodMemory != nil {
+				defaultTaskPodMemory = *currentDeployment.DefaultTaskPodMemory
+			} else {
+				defaultTaskPodMemory = configOption.ResourceQuotas.DefaultPodSize.Memory.Default
+			}
+		}
+		if resourceQuotaCpu == "" {
+			resourceQuotaCpu = *currentDeployment.ResourceQuotaCpu
+		}
+		if resourceQuotaMemory == "" {
+			resourceQuotaMemory = *currentDeployment.ResourceQuotaMemory
+		}
+		if IsDeploymentStandard(*currentDeployment.Type) {
+			var requestedExecutor astroplatformcore.UpdateStandardDeploymentRequestExecutor
+			if executor == "" {
+				requestedExecutor = astroplatformcore.UpdateStandardDeploymentRequestExecutor(*currentDeployment.Executor)
+			}
+			if executor == CeleryExecutor {
+				requestedExecutor = astroplatformcore.CELERY
+			}
+			if executor == KubeExecutor {
+				requestedExecutor = astroplatformcore.KUBERNETES
+			}
+			standardDeploymentRequest := astroplatformcore.UpdateStandardDeploymentRequest{
+				Description:          &description,
+				Name:                 name,
+				Executor:             requestedExecutor,
+				IsCicdEnforced:       isCicdEnforced,
+				IsDagDeployEnabled:   dagDeployEnabled,
+				IsHighAvailability:   highAvailabilityValue,
+				WorkspaceId:          currentDeployment.WorkspaceId,
+				Type:                 astroplatformcore.UpdateStandardDeploymentRequestTypeSTANDARD,
+				ResourceQuotaCpu:     resourceQuotaCpu,
+				ResourceQuotaMemory:  resourceQuotaMemory,
+				EnvironmentVariables: deploymentEnvironmentVariablesRequest,
+				DefaultTaskPodCpu:    defaultTaskPodCpu,
+				DefaultTaskPodMemory: defaultTaskPodMemory,
+			}
+			if schedulerSize != "" {
+				if schedulerSize == SmallScheduler {
+					standardDeploymentRequest.SchedulerSize = astroplatformcore.SMALL
+				}
+				if schedulerSize == MediumScheduler {
+					standardDeploymentRequest.SchedulerSize = astroplatformcore.MEDIUM
+				}
+				if schedulerSize == LargeScheduler {
+					standardDeploymentRequest.SchedulerSize = astroplatformcore.LARGE
+				}
+			}
+			if schedulerSize == "" {
+				standardDeploymentRequest.SchedulerSize = astroplatformcore.UpdateStandardDeploymentRequestSchedulerSize(*currentDeployment.SchedulerSize)
+			}
+			if standardDeploymentRequest.Executor == astroplatformcore.CELERY {
+				if *currentDeployment.Executor == astroplatformcore.DeploymentExecutorKUBERNETES {
+					confirmWithUser = true
+				}
+				if *currentDeployment.Executor == astroplatformcore.DeploymentExecutorKUBERNETES && workerQueuesRequest == nil {
+					standardDeploymentRequest.WorkerQueues = &defautWorkerQueue
+				} else {
+					standardDeploymentRequest.WorkerQueues = &workerQueuesRequest
+				}
+			}
+			if standardDeploymentRequest.Executor == astroplatformcore.KUBERNETES {
+				if *currentDeployment.Executor == astroplatformcore.DeploymentExecutorCELERY {
+					confirmWithUser = true
+				}
+			}
```

Comment:
would simplifying this to something like this work:
```
if standardDeploymentRequest.Executor != *currentDeployment.Executor {
					confirmWithUser = true
				}
			}
```
if this works, we can also remove the check that we had for the vice versa case

Reply:
This doesn't work because astroplatformcore.KUBERNETES and astroplatformcore.DeploymentExecutorCELERY are of different types